ksp_plugin_adapter/window_renderer.cs

```diff
@@ -260,6 +265,41 @@ public interface ISupervisor {
   private readonly ISupervisor supervisor_;
 }
 
+// A supervised window that displays properties of a vessel, given by a
+// delegate.  The delegate must return a nonnull value if and only if there is
+// a currently selected/active vessel *and* it is known to the plugin.
+// Subclasses must access the vessel through the predicted_vessel property and
+// should not cache it.
+internal abstract class
+    VesselSupervisedWindowRenderer : SupervisedWindowRenderer {
+  public delegate Vessel PredictedVessel();
+
+  protected VesselSupervisedWindowRenderer(ISupervisor supervisor,
+                                           PredictedVessel predicted_vessel,
+                                           params UnityEngine.GUILayoutOption[]
+                                               options) : base(
+      supervisor,
+      options) {
+    predicted_vessel_ = predicted_vessel;
+  }
+
+  // A helper for implementing the RenderButton() method of the subclasses.
+  protected void RenderButton(string text) {
```

Comment:
This does not seem specific to vessel-related windows: the plotting frame selectors have a button too.

Should there be `RenderButton` in the base class, overriden here (not necessarily in this pull request, but maybe file a cleanup issue).

Reply:
Look at the code of this function: it is dependent on the existence of a predicted vessel.  That's why it belongs here.  This was horribly broken in the `OrbitAnalyser`, as the decision to hide was made in the middle of `RenderWindow`, a sure recipe for layout/repaint inconsistencies.  This proves that the logic is non-obvious and belongs to a common place.